addons/web/static/tests/helpers/mock_server.js

```diff
@@ -38,7 +38,16 @@ var MockServer = Class.extend({
 
             for (var i = 0; i < model.records.length; i++) {
                 var record = model.records[i];
+                // add potentially missing ids
+                if (record.id === undefined) {
+                    record.id = this._getUnusedID(modelName);
+                }
                 this._applyDefaults(model, record);
+                // ensure initial data goes through proper conversion (x2m, ...)
+                this._mockWrite(modelName, [
+                    [record.id],
+                    record,
+                ]);
```

Comment:
looks like `_mockCreate` without the mandatory `id` to provide

Reply:
Correct, except create doesn't accept `id`, which is a good idea, but it's provided a lot in data at init.